sources/packages/backend/apps/workers/src/controllers/assessment/_tests_/e2e/assessment.controller-verifyAssessmentCalculationOrder.e2e-spec.ts

```diff
@@ -1217,4 +1222,257 @@ describe("AssessmentController(e2e)-verifyAssessmentCalculationOrder", () => {
+            studentAssessmentStatus: StudentAssessmentStatus.Completed,
+          },
+        },
+      );
+      const currentAssessment = createFakeStudentAssessment(
+        {
+          auditUser: student.user,
+          application: pastApplication,
+          offering: pastApplication.currentAssessment.offering,
+        },
+        {
+          initialValue: {
+            assessmentWorkflowId: "some fake id",
+            triggerType: AssessmentTriggerType.ManualReassessment,
+            studentAssessmentStatus: StudentAssessmentStatus.Completed,
+            assessmentDate: addDays(-1),
+          },
+        },
+      );
+      pastApplication.currentAssessment = currentAssessment;
+      await db.application.save(pastApplication);
+
+      const currentAssessmentDisbursement = createFakeDisbursementSchedule(
+        {
+          studentAssessment: currentAssessment,
+          disbursementValues: [
+            createFakeDisbursementValue(
+              DisbursementValueType.CanadaGrant,
+              "CSPT",
+              1150,
+              { disbursedAmountSubtracted: 1000 },
+            ),
+            createFakeDisbursementValue(
+              DisbursementValueType.CanadaGrant,
+              "CSGD",
+              500,
+              { disbursedAmountSubtracted: 500 },
+            ),
+          ],
+        },
+        {
+          initialValues: {
+            coeStatus: COEStatus.required,
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+          },
+        },
+      );
+      currentAssessment.disbursementSchedules = [currentAssessmentDisbursement];
+      await db.studentAssessment.save(currentAssessment);
+
+      // Application currently being processed.
+      const currentApplication = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        { student },
+        {
+          offeringIntensity: OfferingIntensity.partTime,
+          applicationStatus: ApplicationStatus.InProgress,
+          currentAssessmentInitialValues: {
+            assessmentWorkflowId: "some fake id",
+            studentAssessmentStatus: StudentAssessmentStatus.InProgress,
+          },
+        },
+      );
+
+      // Act
+      const result =
+        await assessmentController.verifyAssessmentCalculationOrder(
+          createFakeVerifyAssessmentCalculationOrderPayload(
+            currentApplication.currentAssessment.id,
+          ),
+        );
+
+      // Asserts
+      expect(FakeWorkerJobResult.getResultType(result)).toBe(
+        MockedZeebeJobResult.Complete,
+      );
+      // Expect to consider only the CSGP from the first disbursement
+      // and ignore the second disbursement with the declined COE.
+      expect(FakeWorkerJobResult.getOutputVariables(result)).toStrictEqual({
+        isReadyForCalculation: true,
+        latestCSLPBalance: 0,
+        programYearTotalPartTimeCSPT: 1150,
+        programYearTotalPartTimeCSGD: 500,
```

Comment:
Minor, suggestion, for these scenarios (e.g. `programYearTotalPartTimeCSGD: 500`) would be helpful to have values that would vary between `valueAmount`, `effectiveAmount`, and `programYearTotalPartTimeCSGD`.